src/main/java/com/epam/lowcost/service/implementations/FlightServiceImpl.java

```diff
@@ -89,20 +92,48 @@ private long calculateInitialFlightPriceByDate(long daysBetween, long minPrice)
         long price;
         if (daysBetween > daysNumber) {
             price = minPrice;
-        }
-        else{
-            price = (long) (minPrice + (daysNumber - daysBetween) * (daysNumber - daysBetween) * (minPrice / ((double)daysNumber*daysNumber)));
+        } else {
+            price = (long) (minPrice + (daysNumber - daysBetween) * (daysNumber - daysBetween) * (minPrice / ((double) daysNumber * daysNumber)));
         }
 
         return price;
     }
 
-    private void updateFlightPrice(Flight flight){
+    private void updateFlightPrice(Flight flight) {
         LocalDateTime dateAfter = flight.getDepartureDate();
         LocalDateTime dateBefore = LocalDateTime.now();
         long daysBetween = DAYS.between(dateBefore, dateAfter);
         long minPrice = flight.getInitialPrice();
-        flight.setInitialPrice(calculateInitialFlightPriceByDate(daysBetween, minPrice));
+        long decreaseBusinessPlacesIncreasePrice = 0;
+        long decreaseEconomyPlacesIncreasePrice = 0;
+        long decreaseDaysBetweenIncreasePrice = calculateInitialFlightPriceByDate(daysBetween, minPrice);
+        // if number of free business places less than quarter of plane business places capacity
+        // price rises on quarter of min price
+        if (getNumberOfFreeBusinessPlaces(flight) < flight.getPlane().getBusinessPlacesNumber() / 4.0) {
+            decreaseBusinessPlacesIncreasePrice = (long) (minPrice / 4.0);
```

Comment:
All constants should be moved into separate variables

Reply:
done